xla/service/cpu/onednn_convolution.cc

```diff
@@ -70,8 +77,138 @@ GetKernelConfig<kOnednnConvConfig>(
+                           ? custom_call->shape().tuple_shapes(0)
+                           : custom_call->shape();
+
+  std::vector<Shape> fused_shapes;
+  for (int i = 2; i < operands.size(); ++i) {
+    fused_shapes.push_back(operands[i]->shape());
+  }
+
+  memory::desc input_md = ShapeToMemDesc(input_shape);
+  memory::desc weights_md = ShapeToMemDesc(weight_shape);
+  memory::desc output_md = ShapeToMemDesc(output_shape);
+
+  std::vector<int64_t> inp_perm_axes(conv_config.dims());
+  std::vector<int64_t> ker_perm_axes(conv_config.dims());
+  std::vector<int64_t> out_perm_axes(conv_config.dims());
+
+  int index_i = 0;
+  int index_o = 0;
+  int index_k = 0;
+
+  inp_perm_axes[conv_config.input().data().batch_dim()] = index_i++;
+  out_perm_axes[conv_config.output().data().batch_dim()] = index_o++;
+  ker_perm_axes[conv_config.kernel().filter().output_feature_dim()] = index_k++;
+
+  inp_perm_axes[conv_config.input().data().feature_dim()] = index_i++;
+  out_perm_axes[conv_config.output().data().feature_dim()] = index_o++;
+  ker_perm_axes[conv_config.kernel().filter().input_feature_dim()] = index_k++;
+
+  std::vector<int64_t> inp_dim_axes(
+      conv_config.input().data().spatial_dims().begin(),
+      conv_config.input().data().spatial_dims().end());
+  std::vector<int64_t> ker_dim_axes(
+      conv_config.kernel().filter().spatial_dims().begin(),
+      conv_config.kernel().filter().spatial_dims().end());
+  std::vector<int64_t> out_dim_axes(
+      conv_config.output().data().spatial_dims().begin(),
+      conv_config.output().data().spatial_dims().end());
+
+  for (int64_t& n : inp_dim_axes) inp_perm_axes[--n] = index_i++;
+  for (int64_t& n : ker_dim_axes) ker_perm_axes[--n] = index_k++;
+  for (int64_t& n : out_dim_axes) out_perm_axes[--n] = index_o++;
+
+  memory::dims strides(conv_config.window().strides().begin(),
+                       conv_config.window().strides().end());
+  memory::dims pad_left(conv_config.window().pad_left().begin(),
+                        conv_config.window().pad_left().end());
+  memory::dims pad_right(conv_config.window().pad_right().begin(),
+                         conv_config.window().pad_right().end());
+  memory::dims rhs_dilations(conv_config.window().window_dilations().begin(),
+                             conv_config.window().window_dilations().end());
+
+  for (int64_t& n : strides) n -= 1;
+  for (int64_t& n : pad_left) n -= 1;
+  for (int64_t& n : pad_right) n -= 1;
+  for (int64_t& n : rhs_dilations) n -= 2;
+
+  uint64_t groups = conv_config.feature_groups();
+
+  std::vector<int> inp_axes(inp_perm_axes.begin(), inp_perm_axes.end());
+  std::vector<int> ker_axes(ker_perm_axes.begin(), ker_perm_axes.end());
+  std::vector<int> out_axes(out_perm_axes.begin(), out_perm_axes.end());
+
+  memory::desc new_inp_md = input_md.permute_axes(inp_axes);
+  memory::desc new_ker_md = weights_md.permute_axes(ker_axes);
+  memory::desc new_res_md = output_md.permute_axes(out_axes);
```

Comment:
Nit: Prefer consistent naming.
```suggestion
  memory::desc new_out_md = output_md.permute_axes(out_axes);
```

Reply:
Done